### PR TITLE
166 handle failed deserialization of partial storychunk

### PR DESCRIPTION
**Pull request by fkengun**
Grapher returns an error code to Keeper on failure of story chunk deserialization
Keeper re-stashes the story chunk back to extract queue to try another time later when error code is returned from extraction RPC

### CODE REVIEW

File: ChronoKeeper/StoryChunkExtractorRDMA.cpp [hunk 1]
```diff
@@ -45,6 +43,7 @@ int chronolog::StoryChunkExtractorRDMA::processStoryChunk(StoryChunk*story_chunk
             return chronolog::CL_ERR_UNKNOWN;
         }
         serialized_story_chunk_size = oss.tellp();
+        oss << '\0';
```
Comment (ibrodkin) [inline ChronoKeeper/StoryChunkExtractorRDMA.cpp:46]: why is line 46 needed?

File: ChronoKeeper/StoryChunkExtractorRDMA.cpp [hunk 2]
```diff
@@ -53,7 +52,7 @@ int chronolog::StoryChunkExtractorRDMA::processStoryChunk(StoryChunk*story_chunk
         LOG_DEBUG("[StoryChunkExtractorRDMA] Serialized story chunk size: {}", serialized_story_chunk_size);
 
         std::vector <std::pair <void*, std::size_t>> segments(1);
-        segments[0].first = (void*)(serialized_buf);
+        segments[0].first = (void*)(&serialized_buf[0]);
         segments[0].second = serialized_story_chunk_size + 1;
```
Comment (ibrodkin) [inline ChronoKeeper/StoryChunkExtractorRDMA.cpp:56]: is line 46 a reason for serialized _story_chunk_size+1 ?

**Review by ibrodkin — COMMENTED** [review]
please see a question about serialization

File: ChronoKeeper/StoryChunkExtractorRDMA.cpp (hunk 2 above)
Comment (fkengun) [reply ChronoKeeper/StoryChunkExtractorRDMA.cpp:56]: > is line 46 a reason for serialized _story_chunk_size+1 ?
`serialized_story_chunk_size+1` is copied from Thallium's example code. Yes, line 46 is for this.

**Review by ibrodkin — APPROVED** [review]
please see a question about serialization

**Pull request merged**